plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveTableHandle.java

```diff
@@ -487,4 +343,184 @@ public String toString()
     }
+
+    public static Builder builder()
+    {
+        return new Builder();
+    }
+
+    public static Builder buildFrom(HiveTableHandle table)
+    {
+        return new Builder(table);
+    }
+
+    public static class Builder
+    {
+        private String schemaName;
+        private String tableName;
+        private Optional<Map<String, String>> tableParameters = Optional.empty();
+        private List<HiveColumnHandle> partitionColumns = ImmutableList.of();
+        private List<HiveColumnHandle> dataColumns = ImmutableList.of();
+        private Optional<List<String>> partitionNames = Optional.empty();
+        private Optional<List<HivePartition>> partitions = Optional.empty();
+        private TupleDomain<HiveColumnHandle> compactEffectivePredicate = TupleDomain.all();
+        private TupleDomain<ColumnHandle> enforcedConstraint = TupleDomain.all();
+        private Optional<HiveBucketHandle> bucketHandle = Optional.empty();
+        private Optional<HiveBucketFilter> bucketFilter = Optional.empty();
+        private Optional<List<List<String>>> analyzePartitionValues = Optional.empty();
+        private Set<ColumnHandle> constraintColumns = ImmutableSet.of();
+        private Set<ColumnHandle> projectedColumns = ImmutableSet.of();
+        private AcidTransaction transaction = NO_ACID_TRANSACTION;
+        private boolean recordScannedFiles;
+        private Optional<Long> maxScannedFileSize = Optional.empty();
+
+        private Builder()
+        {
+        }
+
+        private Builder(HiveTableHandle table)
+        {
+            this.schemaName = table.schemaName;
+            this.tableName = table.tableName;
+            this.tableParameters = table.tableParameters;
+            this.partitionColumns = table.partitionColumns;
+            this.dataColumns = table.dataColumns;
+            this.partitionNames = table.partitionNames;
+            this.partitions = table.partitions;
+            this.compactEffectivePredicate = table.compactEffectivePredicate;
+            this.enforcedConstraint = table.enforcedConstraint;
+            this.bucketHandle = table.bucketHandle;
+            this.bucketFilter = table.bucketFilter;
+            this.analyzePartitionValues = table.analyzePartitionValues;
+            this.constraintColumns = table.constraintColumns;
+            this.projectedColumns = table.projectedColumns;
+            this.transaction = table.transaction;
+            this.recordScannedFiles = table.recordScannedFiles;
+            this.maxScannedFileSize = table.maxScannedFileSize;
+        }
+
+        public Builder withSchemaName(String schemaName)
+        {
+            this.schemaName = schemaName;
+            return this;
+        }
+
+        public Builder withTableName(String tableName)
+        {
+            this.tableName = tableName;
+            return this;
+        }
+
+        public Builder withTableParameters(Optional<Map<String, String>> tableParameters)
```

Comment:
Doesn't need to be optional anymore because the default is empty and there is no need to call this to set empty optional.
Similar comment for the other methods with Optional parameter

Reply:
There are some drawbacks to this approach:
1. We'll need to adjust the cases where the caller holds an `Optional` like at `HiveTableHandle#fromJsonForDeserializationOnly` and do something like this:
```
Builder builder = builder()
        .with...
bucketHandle.map(builder::withBucketHandle);  
bucketFilter.map(builder::withBucketFilter);
analyzePartitionValues.map(builder::withAnalyzePartitionValues);
return builder.build();
```
2. We won't be able to use `HiveTableHandle.buildFrom(handle)` and then override those optional values to `Optional.empty`. This is problematic when called outside of Trino and might also be problematic in the connector itself. For example, `maxScannedFileSize` at `HiveMetadata#beginOptimize` and `partitionNames`, `partitionList` at `HivePArtitionManager#applyPartitionResult` (need to check the actual logic to determine if those Optionals could have values and intentionally be overridden to `Optional.empty()`)

WDYT? Can we use the original approach?

Reply:
Right, I think we can keep existing approach